ui-admin/src/forms/designer/ElementList.test.tsx

```diff
@@ -43,7 +43,7 @@ describe('ElementList', () => {
 
     // Act
     const moveBarUpButton = screen.getAllByLabelText('Move this element before the previous one')[1]
-    await user.click(moveBarUpButton)
+    await act(() => user.click(moveBarUpButton))
```

Comment:
I tried to do a little learning about `act()` and most of what I came up with was either [legacy documentation](https://legacy.reactjs.org/docs/test-utils.html#act) or articles ([1](https://javascript.plainenglish.io/you-probably-dont-need-act-in-your-react-tests-2a0bcd2ad65c) [2](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily)) saying that `act()` isn't necessary. Is what I found accurate or is there some other documentation/article you know of that I could learn from?

Reply:
After switching to the Button component, this showed a warning about something not wrapped in `act`. It looked like it was related to the tooltip state. I added this and didn't investigate further.

Reply:
But yeah, generally, act vs no act has been frustrating and often seems inconsistent. I _think_ act is only necessary when the action triggers something async.